arangod/RestServer/QueryRegistryFeature.cpp

```diff
@@ -148,14 +148,14 @@ void QueryRegistryFeature::prepare() {
 
   // create the query registery
   _queryRegistry.reset(new aql::QueryRegistry(_queryRegistryTTL));
-  QUERY_REGISTRY = _queryRegistry.get();
+  QUERY_REGISTRY.store(_queryRegistry.get(), std::memory_order_relaxed);
```

Comment:
This should use `memory_order_release`.

Reply:
I really wasn't sure since we really only set the vars here on startup and shutdown. But I suppose you're technically correct about the memory ordering (as always 👍 )

Reply:
You might be right. If it is guaranteed that the store is performed before any of the other threads is started a relaxed-store would be sufficient. But I would still go with acquire/release in case you change this a some point - it doesn't cost you anything (at least on Intel). Better safe than sorry! :wink:
However, it is perfectly safe to use relaxed-stores to reset the pointers during shutdown.